emu/src/render/ppu.rs

```diff
@@ -0,0 +1,175 @@
+use std::{cell::RefCell, rc::Rc};
+
+#[cfg(feature = "debug")]
+use rand::Rng;
+
+use crate::{
+    memory::internal_memory::InternalMemory,
+    render::{
+        color::{Color, PaletteType},
+        lcd::Lcd,
+        DISPLAY_HEIGHT, DISPLAY_WIDTH, MAX_COLORS_FULL_PALETTE, MAX_COLORS_SINGLE_PALETTE,
+        MAX_PALETTES_BY_TYPE,
+    },
+};
+
+#[derive(Default)]
+pub struct PixelProcessUnit {
+    internal_memory: Rc<RefCell<InternalMemory>>,
+    lcd: Rc<RefCell<Lcd>>,
+}
+
+impl PixelProcessUnit {
+    pub fn new(internal_memory: Rc<RefCell<InternalMemory>>, lcd: Rc<RefCell<Lcd>>) -> Self {
+        Self {
+            internal_memory,
+            lcd,
+        }
+    }
+
+    pub fn render(&self) -> Rc<RefCell<Lcd>> {
+        let gba_display = self.lcd.clone();
+
+        // BG_MODE is forced to 3 to avoid crash on start
```

Comment:
Will it always be like this or this is just for now that we onlt have BG_MODE 3?

Reply:
Just now.